File: server/wrap_7zip.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-

# Copyright (c) 2015, Matthew Brennan Jones <matthew.brennan.jones@gmail.com>
# emu_archive is a HTML based front end for video game console emulators
# It uses a MIT style license
# It is hosted at: https://github.com/workhorsy/emu_archive
#
# Permission is hereby granted, free of charge, to any person obtaining
# a copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, sublicense, and/or sell copies of the Software, and to
# permit persons to whom the Software is furnished to do so, subject to
# the following conditions:
#
# The above copyright notice and this permission notice shall be included
# in all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
# IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
# CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.


import os, sys
import subprocess


class Wrap7zip(object):
	def __init__(self):
		# Check if 7zip is installed
		if os.path.isfile('C:/Program Files/7-Zip/7z.exe'):
			self.exe = 'C:/Program Files/7-Zip/7z.exe'
		elif os.path.isfile('C:/Program Files (x86)/7-Zip/7z.exe'):
			self.exe = 'C:/Program Files (x86)/7-Zip/7z.exe'
		else:
			print('7-Zip wrapper could not locate 7z.exe')
			sys.exit()

	def uncompress(self, comprerssed_file, out_dir):
		command = '"{0}" x -y "{1}" -o{2}'.format(self.exe, comprerssed_file, out_dir)
		proc = subprocess.Popen(command, stdout=subprocess.PIPE)
		out, err = proc.communicate()

File: client/helpers/wrap_7zip.go
// Copyright (c) 2015, Matthew Brennan Jones <matthew.brennan.jones@gmail.com>
// emulators-online is a HTML based front end for video game console emulators
// It uses the GNU AGPL 3 license
// It is hosted at: https://github.com/workhorsy/emulators-online
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

package helpers

import (
	"strings"
	"fmt"
	"os/exec"
	"bytes"
)

func Uncompress(compressed_file string, out_dir string) {
	fmt.Printf("!!!!!!!!!!!!!! uncomressing!\r\n")

	if strings.HasSuffix(compressed_file, ".7z") {
		// Get the command and arguments
		command := "7za.exe"
		args := []string {
			"x",
			"-y",
			fmt.Sprintf(`%s`, compressed_file),
			fmt.Sprintf("-o%s", out_dir),
		}

		// Run the command and wait for it to complete
		cmd := exec.Command(command, args...)
		var stdout bytes.Buffer
		var stderr bytes.Buffer
		cmd.Stdout = &stdout
		cmd.Stderr = &stderr
		err := cmd.Run()
		if err != nil {
			fmt.Printf("Failed to run command: %s\r\n", err)
			fmt.Printf("stdout: %s\r\n", stdout.Bytes())
			fmt.Printf("stderr: %s\r\n", stderr.Bytes())
		}
	} else if strings.HasSuffix(compressed_file, ".rar") {
		
	}
}






